### ISSUE TITLE
ContentEncoding fail for ico file

### ISSUE THREAD

**Issue by iGitScor** [MikaAK/s3-plugin-webpack#66]
# Issue Details

Hi,

thank your for your great work with this plugin !
I have an issue but I don't really know exactly where it fails.
The problem is that some extension does not accept ContentEncoding.

## Environment

Here my environment:
* Docker (Linux 4.9.4-moby x86_64) running on Mac OS 10.12.2
* Node `v6.9.3`
* npm `3.10.10`

Plugin information: `v0.9.2`

## Configuration

The plugin configuration is the following one:

```javascript
common.plugins.push(
    new S3Plugin({
      include: /.*\.(css|js|svg|png|jpe?g|gif|ico|pdf)/,

      s3Options: s3Options,
      s3UploadOptions: {
        Bucket: siteConfiguration.parameters.amazon_bucket,
        /*
         * Images cache-control: 1 day
         * Other resources cache-control: 1 week
         * Data resources cache-control: 1 month with revalidation
         * Scripts and styles cache-control: 1 year
         */
        CacheControl(filename) {
          var cacheControl = configuration.s3CacheControl.misc;
          if (
           configuration.s3CacheControl.imagesExtensions
             .indexOf(path.extname(filename)) >= 0
          ) {
            cacheControl = configuration.s3CacheControl.images;
          }  else if (
            configuration.s3CacheControl.noCacheExtensions
              .indexOf(path.extname(filename)) >= 0
           ) {
            cacheControl = configuration.s3CacheControl.noCache;
          } else if (/\.js/.test(filename) || /\.css/.test(filename)) {
            cacheControl = configuration.s3CacheControl.stylesAndScripts;
          }

          return cacheControl;
        },

        ContentEncoding(filename) {
          // gzip only css and js files
          if (
            /\.js/.test(filename) ||
            /\.css/.test(filename) ||
            /\.ico/.test(filename)
          ) {
            return 'gzip';
          }
        },
      },

      basePathTransform: function () {
        return configuration.s3AssetPath;
      },

      cloudfrontInvalidateOptions:
        siteConfiguration.parameters.amazon_cloudfront_distribution_id ?
        {
          DistributionId: siteConfiguration.parameters.amazon_cloudfront_distribution_id,
          Items: [`/${configuration.s3AssetPath}*`],
        } :
        {},
    })
  );
```

## Debug

I launch webpack as usual and everything worked for me until I've decided to gzip favicon file.
I tried to add a `console.log` in the ContentEncoding condition and the output is:

```
favicon.ico
bundles/mybundle/core.js
bundles/mybundle/helper.js
bundles/mybundle/layout.css
bundles/mybundle/third-party.js
application.js
assets.js
bundler.js
style.css
```

When I check in the S3 console, the metadata is not filled for the favicon but is filled for both css and js files

<img width="543" alt="capture d ecran 2017-02-22 a 14 58 26" src="https://cloud.githubusercontent.com/assets/2276944/23214563/703847e2-f90f-11e6-97a4-49259116d853.png">

Do you need more information ?

Thank you

@iGitScor

**Comment by MikaAK** [comment]
I ran into issues gzipping a favicon so there's an exclude rule built in to prevent it

**Comment by iGitScor** [comment]
Thank you for your answer 😄 

I ignore that files are gzipped using this plugin. I use another one for the compression and the favicon is well compressed. The only problem to display it is that the header content-encoding is not set.

Do you think we should mention this point in the documentation or maybe add option to force the metadata to be pushed ?

**Comment by MikaAK** [comment]
I'll remove the preventing of it in my next update. I forgot why I stopped it 😛

**Comment by MikaAK** [comment]
Fixed in 1.0.1
